### PR TITLE
Adds simple create-delete test for Stream

### PR DESCRIPTION
**Pull request by jaypipes**
In debugging e2e test failure, noted that the `DescribeStreamSummary`
API model does not indicate that `ResourceNotFoundException` is the
404 error... so regenerated the controller after specifying that exception.

Signed-off-by: Jay Pipes <jaypipes@gmail.com>

By submitting this pull request, I confirm that my contribution is made
under the terms of the Apache 2.0 license.

### CODE REVIEW

File: apis/v1alpha1/types.go [hunk 1]
```diff
@@ -55,63 +46,21 @@ type EnhancedMetrics struct {
 	ShardLevelMetrics []*string `json:"shardLevelMetrics,omitempty"`
 }
 
-// The range of possible hash key values for the shard, which is a set of ordered
-// contiguous positive integers.
-type HashKeyRange struct {
-	EndingHashKey   *string `json:"endingHashKey,omitempty"`
-	StartingHashKey *string `json:"startingHashKey,omitempty"`
-}
```
Comment (A-Hilaly) [inline apis/v1alpha1/types.go:63]: I'm wondering what caused all these types to disappear

**Review by A-Hilaly — APPROVED** [review]
👍

**Comment by A-Hilaly** [comment]
@jaypipes we need to add @ack-bot to the maintainers list to run prow jobs for this controllers.

File: apis/v1alpha1/types.go (hunk 1 above)
Comment (jaypipes) [reply apis/v1alpha1/types.go:63]: I regenerated with the latest version of code-gen... not sure :)

**Comment by jaypipes** [comment]
> @jaypipes we need to add @ack-bot to the maintainers list to run prow jobs for this controllers.

@A-Hilaly unfortunately I cannot do that because ack-bot isn't a member of the aws-controllers-k8s Github Organization... If it were a member of that organization, I could add it to the kinesis-maintainer team.

We need to talk to OSPO about getting ack-bot added to the aws-controllers-k8s Github Organization.

**Pull request merged**